File: CS50x/pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
float o;
int q = 25;
int d = 10;
int n = 5;
int p = 1;
int sq = 0;
int sd = 0;
int sn = 0;
int sp = 0;

    do
    {
       printf("O hai! How much change is owed?\n"); 
       o = get_float();
    }
    while(o < 0);

float (x) = o * 100; 
int c = round(x);

    if (c >= q)
    {
        sq = c / q;
        c = c % q;
    }
    if (c >= d)
    {
        sd = c / d;
        c = c % d;
    }
    if(c >= n)
    {
        sn = c / n;
        c = c % n;
    }
    if(c >= p)
    {
        sp = c / p;
    }
    
int t = sq + sd + sn + sp;
printf("%i\n", t);
    
}

File: CS50x/pset6/greedy.py
import cs50

q = 25
d = 10
n = 5
p = 1

sq = 0
sd = 0
sn = 0
sp = 0

while True:
    print("O hai! How much change is owed?")
    o = cs50.get_float()
    if o > 0:
        break

x = o * 100;
c = round(x)

if c >= q:
    sq = c // q;
    c = c % q
    
if c >= d:
    sd = c // d
    c = c % d
    
if c >= n:
    sn = c / n
    c = c % n 
    
if c >= p:
    sp = c / p
    
t = sq + sd + sn + sp
print(int(t))

